build.sbt

```
@@ -163,3 +163,11 @@ lazy val xml = crossProject(JSPlatform, JVMPlatform)
     Test / fork := false
   )
   .jsConfigure(_.enablePlugins(ScalaJSJUnitPlugin))
+  .nativeSettings(
+    scalaModuleMimaPreviousVersion := None, // No such release yet
+    // Scala Native cannot run forked tests
+    Test / fork := false,
+    Test / libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion,
```

This should fix the compile time errors. Everything other LGTM, although I'm courious were this test should be finally placed. CircleCI has also configuration for ScalaJs, but not for ScalaNative
```suggestion
    Test / libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion,
    libraryDependencies += "com.novocode"  % "junit-interface" % "0.11" % Test,
```

---

Good to know! Let me try!

---

@WojciechMazur Nope, it doesn't work if I move the dependencies to the Test scope, even adding `"com.novocode"  % "junit-interface" % "0.11" % Test`

---

Ok, so the problem is we're adding  native junit compilert plugin to the whole project, so even if we try to execute compilaton on standard sources it still will be trigered. 
We forget to add something similar to `ScalaJSJUnitPlugin` that would add compiler junit plugin only when compiling tests. 
I'm not sure, but probably for now the only option of adding `scala-native-junit-compiler-plugin` would be to manually add it to classpath ([here's how it's done in ScalaJs](https://github.com/scala-js/scala-js/blob/master/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSJUnitPlugin.scala) ).

---

It should just be
```suggestion
    libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion % Test,
```
`Test / libraryDependencies` is never used by sbt.

---

But yes, it's true that the plugin should only be added in the `Test` configuration as well. This is not a feature that sbt offers by default. It can be done like we do in Scala.js, with a custom config. Or since sbt 1.x it's also possible to directly use the `librarymanagement` API, which might be easier. It should look something like
```scala
val lm = dependencyResolution.value
val cp = lm
  .retrieve("org.scala-native" % "junit-plugin_${scalaVersion.value}" % scalaNativeVersion, scalaModuleInfo = None, retrieveDir, log)
  .fold(w => throw w.resolveException, Attributed.blankSeq(_))
cp.find(... ~ "junit-plugin")
```

---

@sjrd It doesn't work either. I get:
```
[error] scala.reflect.internal.MissingRequirementError: object org.junit.Test in compiler mirror not found.
```
@WojciechMazur I guess it's fine to depend on it in the compile scope ( it's what other libraries using JUnit are doing ) and change once the issue is fixed in a future Scala Native version?

---

> I guess it's fine to depend on it in the compile scope

If you do that, then everyone who depends on your library gets JUnit on their classpath. We should instead enable the plugin only in the Test config.

---

@sjrd Tried what you suggested. Can I ask you another look? :)

---

- https://github.com/scala-native/scala-native/issues/2540